local_planner/src/local_planner.cpp

```diff
@@ -106,6 +106,9 @@ LocalPlanner::LocalPlanner(ros::NodeHandle nh)
 
   // Initialize the plan index
   current_plan_index_ = 0;
+
+  // Initialized reference plan for yaw
+  first_ref_plan = false;
```

Comment:
Hmm... that is weird. I believe the member variable first_ref_plan should be initialized in the local_planner.h. Also, added a second commit and corrected a small problem.  first_ref_plan should be defined as true when first defined.

Reply:
Hi Andrew, sorry for the delay in the response. 
Could you quantify the improvement on the simulation by applying this wrapping?
Have you noticed if the issue with the angle still shows up? As I tested, there is times in which MPC still fails.

Reply:
Hey Selvin,

Sorry for the false positive. I have tested this more extensively, and have noticed that this may be the case. It was able to solve going backwards in the scenery shown in issue #204, but it seems that there are still some issues. I won't be able to get towards debugging this unwrapping until a couple weeks later as I am preparing to attend a conference soon. 

I will try to debug this a little bit more extensively once I get back. I will most likely generate a trajectory with an offline planner module that I have to get a good understanding of why this unwrapping is not working for all cases.

Reply:
Hey Andrew, sounds good. In that case, we can leave this open for now until you get to it. Please ping us with any updates and thank you for contributing to the codebase. Solving this issue is also important for us so appreciate your efforts.